lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -2,11 +2,30 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class ResourceIdentifier
+        def self.type_for(class_name, serializer_type = nil, transform_options = {})
+          if serializer_type
+            raw_type = serializer_type
+          else
+            inflection =
+              if ActiveModelSerializers.config.jsonapi_resource_type == :singular
```

Comment:
Looks like pluralize is used in

lib/active_model/serializer/collection_serializer.rb
lib/active_model/serializer/fieldset.rb
lib/active_model_serializers/adapter.rb
lib/active_model_serializers/adapter/json_api/resource_identifier.rb

and singularize in 

lib/active_model/serializer/fieldset.rb
lib/active_model_serializers/adapter/json_api/deserialization.rb
lib/active_model_serializers/adapter/json_api/resource_identifier.rb

String.underscore is called in

lib/active_model/serializer.rb
lib/active_model/serializer/collection_serializer.rb
lib/active_model_serializers/adapter.rb
lib/active_model_serializers/adapter/json_api/deserialization.rb
lib/active_model_serializers/adapter/json_api/resource_identifier.rb

could use an audit